pairviewer.js

```diff
@@ -1263,14 +1265,50 @@ function fadeAtEdge(d) {
     end = d.coordinates2;
   }
 
+  distancePair = d3.geoDistance(start, end);
+
   let start_dist = 1.57 - d3.geoDistance(start, centerPos),
     end_dist = 1.57 - d3.geoDistance(end, centerPos);
 
   let fade = d3
     .scaleLinear()
     .domain([-0.1, 0])
     .range([0, 0.15]);
+
   let dist = start_dist < end_dist ? start_dist : end_dist;
+
+  if(distancePair>=1.7) {
+    // 1.7 radians has been taken as an approximate value for a "long" flyer.
+    // However, this value can be changed as desired.
+    let avgLongitudeStartEnd = (start[0]+end[0])/2, // average longitude of the flyer's start and end points
+    avgLatitudeStartEnd = (start[1]+end[1])/2,      // average latitude of the flyer's start and end points
+    diffAvgCenterLongitude,                         // difference between avgLongitudeStartEnd and the longitude in centerPos
+    diffAvgCenterLatitude,                          // difference between avgLatitudeStartEnd and the latitude in centerPos
+    atBack = false;                                 // boolean representing if the flyer is at the back of the globe
+
+    diffAvgCenterLongitude = avgLongitudeStartEnd - centerPos[0];
+    if (diffAvgCenterLongitude > 180) diffAvgCenterLongitude = -180 + (diffAvgCenterLongitude-180);
+    else if (diffAvgCenterLongitude < 180) diffAvgCenterLongitude = 180 - (diffAvgCenterLongitude+180);
+
+    diffAvgCenterLatitude = avgLatitudeStartEnd - centerPos[1];
+    if (diffAvgCenterLatitude > 180) diffAvgCenterLatitude = -180 + (diffAvgCenterLatitude-180);
```

Comment:
Can you format this into multiple lines (same with the other places too):
`if (....) { `
    `....`
`}`